L1Trigger/DTTriggerPhase2/src/MPCleanHitsFilter.cc

```diff
@@ -0,0 +1,56 @@
+#include "L1Trigger/DTTriggerPhase2/interface/MPCleanHitsFilter.h"
+
+using namespace edm;
+using namespace std;
+
+// ============================================================================
+// Constructors and destructor
+// ============================================================================
+MPCleanHitsFilter::MPCleanHitsFilter(const ParameterSet &pset) : MPFilter(pset) {
+  // Obtention of parameters
+  debug_ = pset.getUntrackedParameter<bool>("debug");
+
+  timeTolerance_ = pset.exists("timeTolerance") ? pset.getParameter<int>("timeTolerance") : 999999;
```

Comment:
Can you please be a bit more explicit about what should be done? I dont manage to dd a fillDescription because it fails with the edm::FileInPath, and I also do not see how it would solve the problem.

Reply:
I managed to generate it through `edmPythonConfigToCppValidation` (see [here](https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideConfigurationValidationAndHelp#Converting_an_Existing_CFI_file) for reference). I suppose you are getting some errors because the path in `edm::FileInPath` is defined with `'` and not `"`.  It would solve the issue because using the `fillDescription` (as I configured below) you could get rid `python/dtTriggerPhase2PrimitiveDigis_cfi.py` and you would always have the `timeTolerance` (that I don't see anyway changed o defined anywhere else in the release) always set by default to `99999` if it is not explicitly changed by a config. This would have de facto the same behavior as the `exists` here. 

I'm adding the `fillDescription` here for reference.


```C++
void
DTTrigPhase2Prod::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
  // dtTriggerPhase2PrimitiveDigis
  edm::ParameterSetDescription desc;
  desc.add<edm::InputTag>("digiTag", edm::InputTag("CalibratedDigis"));
  desc.add<int>("timeTolerance", 999999);
  desc.addUntracked<std::string>("geometry_tag", "");
  desc.addUntracked<int>("trigger_with_sl", 4);
  desc.addUntracked<double>("tanPhiTh", 1.0);
  desc.addUntracked<double>("tanPhiThw2max", 1.3);
  desc.addUntracked<double>("tanPhiThw2min", 0.5);
  desc.addUntracked<double>("tanPhiThw1max", 0.9);
  desc.addUntracked<double>("tanPhiThw1min", 0.2);
  desc.addUntracked<double>("tanPhiThw0", 0.5);
  desc.addUntracked<double>("chi2Th", 0.01);
  desc.addUntracked<double>("chi2corTh", 0.1);
  desc.add<bool>("do_correlation", true);
  desc.addUntracked<bool>("useBX_correlation", false);
  desc.addUntracked<double>("dT0_correlate_TP", 25.0);
  desc.addUntracked<int>("dBX_correlate_TP", 0);
  desc.addUntracked<double>("dTanPsi_correlate_TP", 99999.0);
  desc.addUntracked<bool>("clean_chi2_correlation", true);
  desc.addUntracked<bool>("allow_confirmation", true);
  desc.addUntracked<double>("minx_match_2digis", 1.0);
  desc.add<int>("scenario", 0);
  desc.add<int>("df_extended", 0);
  desc.addUntracked<bool>("filter_cousins", true);
  desc.add<int>("max_primitives", 999);
  desc.add<edm::FileInPath>("ttrig_filename", edm::FileInPath("L1Trigger/DTTriggerPhase2/data/wire_rawId_ttrig.txt"));
  desc.add<edm::FileInPath>("z_filename", edm::FileInPath("L1Trigger/DTTriggerPhase2/data/wire_rawId_z.txt"));
  desc.add<edm::FileInPath>("shift_filename", edm::FileInPath("L1Trigger/DTTriggerPhase2/data/wire_rawId_x.txt"));
  desc.add<edm::FileInPath>("shift_theta_filename", edm::FileInPath("L1Trigger/DTTriggerPhase2/data/theta_shift.txt"));
  desc.add<edm::FileInPath>("global_coords_filename", edm::FileInPath("L1Trigger/DTTriggerPhase2/data/global_coord_perp_x_phi0.txt"));
  desc.add<int>("algo", 0);
  desc.addUntracked<int>("minHits4Fit", 3);
  desc.addUntracked<bool>("splitPathPerSL", true);
  desc.addUntracked<bool>("debug", false);
  desc.addUntracked<bool>("dump", false);
  desc.add<edm::InputTag>("rpcRecHits", edm::InputTag("rpcRecHits"));
  desc.add<bool>("useRPC", false);
  desc.addUntracked<int>("bx_window", 1);
  desc.addUntracked<double>("phi_window", 50.0);
  desc.addUntracked<int>("max_quality_to_overwrite_t0", 9);
  desc.addUntracked<bool>("storeAllRPCHits", false);
  desc.add<bool>("activateBuffer", false);
  desc.add<double>("superCelltimewidth", 400);
  desc.add<int>("superCellspacewidth", 20);
  {
    edm::ParameterSetDescription psd0;
    psd0.addUntracked<bool>("debug", false);
    psd0.addUntracked<double>("angletan", 0.3);
    psd0.addUntracked<double>("anglebinwidth", 1.0);
    psd0.addUntracked<double>("posbinwidth", 2.1);
    psd0.addUntracked<double>("maxdeltaAngDeg", 10);
    psd0.addUntracked<double>("maxdeltaPos", 10);
    psd0.addUntracked<int>("UpperNumber", 6);
    psd0.addUntracked<int>("LowerNumber", 4);
    psd0.addUntracked<double>("MaxDistanceToWire", 0.03);
    psd0.addUntracked<int>("minNLayerHits", 6);
    psd0.addUntracked<int>("minSingleSLHitsMax", 3);
    psd0.addUntracked<int>("minSingleSLHitsMin", 3);
    psd0.addUntracked<bool>("allowUncorrelatedPatterns", true);
    psd0.addUntracked<int>("minUncorrelatedHits", 3);
    desc.add<edm::ParameterSetDescription>("HoughGrouping", psd0);
  }
  {
    edm::ParameterSetDescription psd0;
    psd0.addUntracked<edm::FileInPath>("pattern_filename", edm::FileInPath("L1Trigger/DTTriggerPhase2/data/PseudoBayesPatterns_uncorrelated_v0.root"));
    psd0.addUntracked<bool>("debug", false);
    psd0.addUntracked<int>("minNLayerHits", 3);
    psd0.addUntracked<int>("minSingleSLHitsMax", 3);
    psd0.addUntracked<int>("minSingleSLHitsMin", 0);
    psd0.addUntracked<int>("allowedVariance", 1);
    psd0.addUntracked<bool>("allowDuplicates", false);
    psd0.addUntracked<bool>("setLateralities", true);
    psd0.addUntracked<bool>("allowUncorrelatedPatterns", true);
    psd0.addUntracked<int>("minUncorrelatedHits", 3);
    psd0.addUntracked<bool>("saveOnPlace", true);
    psd0.addUntracked<int>("maxPathsPerMatch", 256);
    desc.add<edm::ParameterSetDescription>("PseudoBayesPattern", psd0);
  }
  descriptions.add("dtTriggerPhase2PrimitiveDigis", desc);
  // or use the following to generate the label from the module's C++ type
  //descriptions.addWithDefaultLabel(desc);
}

```